apps/opik-backend/src/main/java/com/comet/opik/domain/LlmProviderApiKeyService.java

```diff
@@ -123,6 +131,32 @@ public void updateApiKey(@NonNull UUID id, @NonNull ProviderApiKeyUpdate provide
         });
     }
 
+    @Override
+    public void delete(UUID id, String workspaceId) {
+        template.inTransaction(WRITE, handle -> {
+
+            var repository = handle.attach(LlmProviderApiKeyDAO.class);
+
+            repository.delete(id, workspaceId);
+
+            // Void return
+            return null;
+        });
+    }
+
+    @Override
+    public void delete(Set<UUID> ids, String workspaceId) {
```

Comment:
Minor: same, validate arguments, missing non-null handling on the set.

Reply:
done